packages/ledger-icp/src/ledger.canister.ts

```diff
@@ -158,4 +162,30 @@ export class LedgerCanister extends Canister<LedgerService> {
 
     return response.Ok;
   };
+
+  /**
+   * Fetches the consent message for a specified canister call, intended to provide a human-readable message that helps users make informed decisions.
+   *
+   * Reference: https://github.com/dfinity/wg-identity-authentication/blob/main/topics/ICRC-21/icrc_21_consent_msg.md
+   *
+   * @param {Icrc21ConsentMessageRequest} params - The request parameters containing the method name, arguments, and consent preferences (e.g., language).
+   * @returns {Promise<icrc21_consent_message_response>} - A promise that resolves to the consent message response, which includes the consent message in the specified language and other related information.
```

Comment:
I would also add `@throws {GenericError | InsufficientPaymentError | UnsupportedCanisterCallError | ConsentMessageUnavailable | ConsentMessageError} - In case the call doesn't return the consent message, the method will raise an error.`